include/xtypes/Module.hpp

```diff
@@ -0,0 +1,404 @@
+        if (module.first == nullptr)
+        {
+            // This will fail
+            return static_cast<StructType&>(const_cast<DynamicType&>(*structs.end()->second));
+        }
+
+        auto it = module.first->structs.find(module.second);
+        if (it != module.first->structs.end())
+        {
+            return static_cast<StructType&>(const_cast<DynamicType&>(*it->second));
+        }
+        // This will fail
+        return static_cast<StructType&>(const_cast<DynamicType&>(*structs.end()->second));
+    }
+
+    bool set_struct(
+            const StructType& struct_type)
+    {
+        if (struct_type.name().find("::") != std::string::npos)
+        {
+            return false; // Cannot add a symbol with scoped name.
+        }
+
+        auto result = structs.emplace(struct_type.name(), struct_type);
+        return result.second;
+    }
+
+    bool set_struct(
+            StructType&& struct_type,
+            bool replace = false)
+    {
+        if (struct_type.name().find("::") != std::string::npos)
+        {
+            return false; // Cannot add a symbol with scoped name.
+        }
+
+        if (replace)
+        {
+            auto it = structs.find(struct_type.name());
+            if (it != structs.end())
+            {
+                structs.erase(it);
+            }
+        }
+
+        auto result = structs.emplace(struct_type.name(), std::move(struct_type));
+        return result.second;
+    }
+
+    // TODO has, get and set of:
+    // enums, bitmasks and unions
+
+    std::map<std::string, DynamicType::Ptr> get_all_types() const
+    {
+        std::map<std::string, DynamicType::Ptr> result;
+        fill_all_types(result);
+        return result;
+    }
+
+    void fill_all_types(
+            std::map<std::string, DynamicType::Ptr>& map) const
+    {
+        map.insert(structs.begin(), structs.end());
+        for (const auto& pair : inner)
+        {
+            pair.second->fill_all_types(map);
+        }
+    }
+
+    DynamicData get_constant(
+            const std::string& name) const
+    {
+        // Solve scope
+        PairModuleSymbol module = resolve_scope(name);
+        if (module.first == nullptr)
+        {
+            return DynamicData(primitive_type<bool>());
+        }
+
+        auto it = module.first->constants.find(module.second);
+        if (it != module.first->constants.end())
+        {
+            return it->second;
+        }
+
+        return DynamicData(primitive_type<bool>());
+    }
+
+    bool has_constant(
+            const std::string& name) const
+    {
+        // Solve scope
+        PairModuleSymbol module = resolve_scope(name);
+        if (module.first == nullptr)
+        {
+            return false;
+        }
+
+        auto it = module.first->constants.find(module.second);
+        if (it != module.first->constants.end())
+        {
+            return true;
+        }
+
+        return false;
+    }
+
+    bool set_constant(
```

Comment:
`add`?

Reply:
Just `constant`?

Reply:
If it is not idempotent I would put `add_constant`. If one overrides the previous one, I left as `.constan`

Reply:
Replaced by `create_constant` and added an optional argument for replacing.